src/apps/wallet/sign_tx/layout.py

```diff
@@ -8,11 +11,27 @@
 from apps.common.confirm import confirm, hold_to_confirm
 from apps.wallet.sign_tx import addresses, omni
 
+# embedded ports of micropython start their epoch 1.1.2000
+# see https://docs.micropython.org/en/latest/library/utime.html
+_EMBEDDED_TIMESTAMP_DIFF = const(946684800)
+
 
 def format_coin_amount(amount, coin):
     return "%s %s" % (format_amount(amount, 8), coin.coin_shortcut)
 
 
+def convert_to_embedded_timestamp(unix_timestamp):
+    return max(0, unix_timestamp - _EMBEDDED_TIMESTAMP_DIFF)
+
+
+# dates before 1.1.2000 will resolve to 1.1.2000 on non-unix
+def format_date(timestamp):
+    if not sys.platform.equals("unix"):
+        timestamp = convert_to_embedded_timestamp(timestamp)
+
+    return utime.localtime(timestamp)
```

Comment:
We might consider avoiding using the `utime` module completely and compute the date in pure Python instead.

Reply:
based on what we agreed in meeting all timestamp->formatted date functionality was removed.